src/http/httpd.cc

```diff
@@ -220,6 +220,10 @@ future<> connection::read_one() {
         }
         ++_server._requests_served;
         std::unique_ptr<http::request> req = _parser.get_parsed_request();
+
+        req->_server_address = _fd.local_address();
+        req->_client_address = _fd.remote_address();
```

Comment:
This happens per every request sent by the client, and both `::getpeername()` and `::getsockname()` are system calls that (!) can yield in the kernel. We typically try to avoid both in seastar. So it's definitely better for `connection::read_one()` to get the local and remote addresses in advance, keep them on connection itself and just copy to request, instead of talking to the kernel every time.

Another thing to note here, is that remote address can be known from the `accept_result` when the new connection is accepted, so the new `remote_address()` call is good to have, but likely not required here.

Reply:
Thanks for the review!  Appreciate the feedback.

> Another thing to note here, is that remote address can be known from the accept_result when the new connection is accepted, so the new remote_address() call is good to have, but likely not required here.

I noticed that one of the constructors for `http::connection` would accept a `socket_address` but was marked deprecated: https://github.com/scylladb/seastar/blob/f11f1b4370ed4414af552e5269bb173c721617fa/include/seastar/http/httpd.hh#L88-L90

Can I un-deprecate this in order to pass the address to the constructor?  I'm not sure what historical reason there is as to why it was deprecated.

Reply:
But you'll need to construct it with _two_ socket_address-es :) So I would say it's not going to be undeprecation. Instead, there will be another constructor with local and remote addresses, and the existing one ... would need to call `_fd.get_local_address()` and `_fd.get_remote_address()` for those values